File: orquestator/handlers/answer.go
package handlers

import (
	"log"
	"net/http"
	"orquestator/actions"
	"orquestator/api/session"

	"github.com/gin-gonic/gin"
)

func AnswerHandler(context *gin.Context) {
	callSid := context.PostForm("CallSid")

	response, err := session.Start(callSid)
	if err != nil {
		log.Println("Answer error:", err)
		return
	}

	if response.Action != "TALK" {
		log.Printf("Session error: Triying to start the conversation with action (%s)\n", response.Action)
		return
	}

	result, err := actions.HandleTalk(callSid, response.Data)
	if err != nil {
		log.Println("Error handling action:", response.Action, err)
		context.String(http.StatusInternalServerError, err.Error())
		return
	}

	context.Header("Content-Type", "text/xml")
	context.String(http.StatusOK, result)
}

File: orquestator/handlers/answer.py
from fastapi import APIRouter, Form
from fastapi.responses import Response
import logging

from api.session import start_session
from actions.registry import get_action_handler

router = APIRouter()
logger = logging.getLogger(__name__)

@router.post("/answer")
async def answer_handler(
    From: str = Form(...),
    CallSid: str = Form(...)
):
    try:
        response = await start_session(CallSid)
        
        if response["action"] != "TALK":
            logger.error(f"Session error: Trying to start with action {response['action']}")
            return Response(content="", media_type="text/xml")
        
        handler = get_action_handler("TALK")
        result = await handler(CallSid, From, response["data"])
        
        return Response(content=result, media_type="text/xml")
    
    except Exception as e:
        logger.error(f"Answer error: {e}")
        return Response(content="", media_type="text/xml")